.github/workflows/pr-actions.yaml

```diff
@@ -236,13 +236,73 @@ jobs:
       overlay-repo: ${{ needs.prepare.outputs.overlay-repo }}
       workspace-path: ${{ needs.prepare.outputs.workspace }}
 
+  add_compatibility_failure_comment:
+    needs:
+      - parse
+      - prepare
+      - checkBackstageCompatibility
+
+    concurrency:
+      group: add_compatibility_failure_comment-${{ github.ref_name }}-${{ github.event.issue.number }}
+      cancel-in-progress: false
+
+    permissions:
+      pull-requests: write
+
+    if: failure() && needs.parse.outputs.command_name == 'publish' && needs.prepare.outputs.overlay-branch != '' && needs.prepare.outputs.workspace != ''
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download compatibility report
+        uses: actions/download-artifact@v4
+        with:
+          name: backstage-compatibility-report
+          path: ./
+
+      - name: Verify compatibility report
+        run: |
+          if [ -f "backstage-compatibility-report.md" ]; then
+            echo "Successfully downloaded backstage-compatibility-report.md"
+          else
+            echo "backstage-compatibility-report.md not found, skipping comment"
+            exit 0
+          fi
+
+      - name: Add compatibility failure comment
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
+            const reportPath = 'backstage-compatibility-report.md';
+
+            if (!fs.existsSync(reportPath)) {
+              console.log('Compatibility report not found, skipping comment');
+              return;
+            }
+
+            const reportContent = fs.readFileSync(reportPath, 'utf8');
+
+            const body = '## Backstage Compatibility Check Failed\n\n' +
```

Comment:
If you add the compatibility check markdown content as part of the `add-publish-completion-comment` job, and only as a followup of the completion comment on failure of the Check job, then you probably don't need to add this title. Pasting the content of the markdown file would be sufficient I guess.